charts/render.py

```diff
@@ -114,25 +120,45 @@ def keep_doc(doc):
 
 
 def main():
-    (name, namespace, values, path) = sys.argv[1:]
+    parser = argparse.ArgumentParser()
+    parser.add_argument('name', help="Denotes the name of the chart")
+    parser.add_argument('namespace', help="Metalk8s namespace to deploy unto")
+    parser.add_argument('values', help="Our custom chart values")
+    parser.add_argument(
+        '--service-config',
+        action='append',
+        nargs=2,
+        dest="service_config",
+        help="Ex: --service-config service_name service_configmap_name"
+    )
```

Comment:
Cool stuff! Could add a small explanation about what those "service-config"s are in the help message (and adjust the example to make it understandable that `service_name` is actually the Jinja variable which will hold this ConfigMap's contents).

(minor) To make it even friendlier to use/read in code, you can cast parsed arguments into arbitrary types:

```python
from collections import namedtuple

ServiceConfig = namedtuple('ServiceConfig', ('var_name', 'configmap_name')

# When building the parser...
parser.add_argument(
    '--service-config',
    action='append',
    nargs=2,
    type=ServiceConfig,
    dest="service_configs",
    help="...",
)

# When printing the Jinja declarations...
sys.stdout.write(
    BOILERPLATE.format(
        configlines='\n'.join(
            ("{{%- set {s.var_name} = salt.metalk8s_service_configurator"
             ".get_service_conf('{namespace}', '{s.configmap_name}') %}}").format(
                 s=service_config, namespace=args.namespace
            ) for service_config in args.service_configs
        )
    ).lstrip()
)
```

Note that you don't need to inject curly braces with `format` arguments, simply double them (`{{` or `}}`) to escape the characters. (cf. https://docs.python.org/library/string.html#formatstrings)